src/core/account.ts

```diff
@@ -169,19 +204,34 @@ export class Account {
   /**
    * Derives an account with bip44 path and mnemonics,
    *
-   * @param args.path the BIP44 derive path (e.g. m/44'/637'/0'/0'/0')
+   * @param args.scheme The signing scheme to derive with
+   * @param args.path the BIP44 derive hardened path (e.g. m/44'/637'/0'/0'/0') for Ed25519,
+   * or non-hardened path (e.g. m/44'/637'/0'/0/0) for secp256k1
    * Detailed description: {@link https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki}
    * @param args.mnemonic the mnemonic seed phrase of the account
-   * @returns AptosAccount
+   * @param args.legacy optional. To indicate whether to use a legacy Ed25519
+   *
+   * @returns Account
    */
-  static fromDerivationPath(args: { path: string; mnemonic: string }): Account {
-    const { path, mnemonic } = args;
-    const { key } = derivePrivateKeyFromMnemonic(KeyType.ED25519, path, mnemonic);
-    const privateKey = new Ed25519PrivateKey(key);
-    const publicKey = privateKey.publicKey();
-    const authKey = Account.authKey({ publicKey });
-    const address = new AccountAddress({ data: authKey.toUint8Array() });
-    return new Account({ privateKey, address, legacy: true });
+  static fromDerivationPath(args: {
+    scheme: SigningSchemeInput;
+    path: string;
+    mnemonic: string;
+    legacy?: boolean;
+  }): Account {
+    const { path, mnemonic, scheme, legacy } = args;
+    let privateKey;
+    switch (scheme) {
+      case SigningSchemeInput.Secp256k1Ecdsa:
+        privateKey = new Secp256k1PrivateKey(Secp256k1PrivateKey.fromDerivationPath(path, mnemonic));
+        break;
+      case SigningSchemeInput.Ed25519:
+        privateKey = new Ed25519PrivateKey(Ed25519PrivateKey.fromDerivationPath(path, mnemonic));
+        break;
+      default:
+        throw new Error(`Unsupported private key ${privateKey}`);
```

Comment:
```suggestion
        throw new Error(`Unsupported scheme ${scheme}`);
```

should this be scheme instead of private key?